AssetRipper.GUI/ConsoleOptions.cs

```diff
@@ -0,0 +1,350 @@
+		/// </summary>
+		/// <typeparam name="T">The type of the resulting object</typeparam>
+		/// <param name="option">An Argument object from System.CommandLine</param>
+		/// <param name="fallback">The value to return if the parse result is null</param>
+		/// <returns>T</returns>
+		public static T GetArgumentOrFallback<T>(Argument<T> argument, T fallback)
+		{
+			if (_context == null)
+			{
+#if DEBUG
+				Console.WriteLine("Programming error: Context was not set before trying to get an argument from it. Returning fallback.");
+#endif
+				return fallback;
+			}
+
+			T? result = _context.ParseResult.GetValueForArgument<T>(argument);
+
+			if (result == null)
+				return fallback;
+
+			return result;
+		}
+
+		#endregion
+
+		#region Options and Arguments
+
+		private static Argument<List<string>>? _inputArgument;
+
+		public static Argument<List<string>> inputArgument
+		{
+			get
+			{
+				if (_inputArgument != null)
+					return _inputArgument;
+
+				Argument<List<string>> argument = new Argument<List<string>>(name: "input", description: "Input files or directory to export");
+
+				argument.Arity = ArgumentArity.OneOrMore;
+				argument.AddValidator((symbolResult) =>
+				{
+					List<string>? inputs = symbolResult.GetValueForArgument(argument);
+
+					foreach (string input in inputs)
+					{
+						string path = ExecutingDirectory.Combine(input);
+
+						if (!MultiFileStream.Exists(path) && !Directory.Exists(path))
+						{
+							symbolResult.ErrorMessage = MultiFileStream.IsMultiFile(path)
+								? $"File '{path}' doesn't have all parts for combining"
+								: $"Neither file nor directory with path '{path}' exists";
+
+							Environment.ExitCode = (int)ExitCode.ArgumentError;
+							return;
+						}
+					}
+				});
+
+				_inputArgument = argument;
+
+				return argument;
+			}
+		}
+
+		private static Option<List<string>>? _inputOption;
+
+		public static Option<List<string>> inputOption
+		{
+			get
+			{
+				if (_inputOption != null)
+					return _inputOption;
+
+				Option<List<string>> option = new Option<List<string>>(name: "--input", description: "Input files or directory to export");
+				option.AddAlias("-i");
+				option.IsRequired = true;
+				option.Arity = ArgumentArity.OneOrMore;
+
+				option.AddValidator((symbolResult) =>
+				{
+					List<string>? inputs = symbolResult.GetValueForOption(option);
+
+					if (inputs == null)
+						return;
+
+					foreach (string input in inputs)
+					{
+						string path = ExecutingDirectory.Combine(input);
+
+						if (!MultiFileStream.Exists(path) && !Directory.Exists(path))
+						{
+							symbolResult.ErrorMessage = MultiFileStream.IsMultiFile(path)
+								? $"File '{path}' doesn't have all parts for combining"
+								: $"Neither file nor directory with path '{path}' exists";
+
+							Environment.ExitCode = (int)ExitCode.ArgumentError;
+							return;
+						}
+					}
+				});
+
+				_inputOption = option;
+
+				return option;
+			}
+		}
```

Comment:
Why is there an `--input` option at all? This was an implicit option before, where anything not directly following an option was considered to be part of the input path list.

Reply:
This option is part of the `extract` subcommand, the root command doesn't use it. `--extract` does not use positional arguments since if there is one, the root command will run. In shell scripts, positional arguments are much less readable with no other benefit since all options are passed in using the command line anyway.